lib/web/server.cpp

```diff
@@ -0,0 +1,719 @@
+#include <iostream>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <unistd.h>
+#include <cstring>
+#include <regex>
+#include <fstream>
+#include <sstream>
+#include <regex>
+#include <iomanip>
+#include <sys/wait.h>
+
+#include <openssl/ssl.h>
+#include <openssl/sha.h>
+
+#include "server.h"
+#include "../util.h"
+
+#define STDIN 0
+#define STDOUT 1
+
+using std::string;
+using std::regex;
+using std::smatch;
+using std::cout;
+using std::to_string;
+using std::map;
+using Socialite::Util::toHttpTimestamp;
+using Socialite::Util::parseHttpTimestamp;
+using Socialite::Util::sha256;
+using Socialite::Util::mapPostData;
+
+Socialite::Web::Server::Server(string cert, string key) : Socialite::Server(443, cert, key, WS_MAX_CLIENT, S_SSL_SERVER) {
+	cout << "New secure Web::Server starting\n";
+	useSSLAlso = false;
+	readConfigFile();
```

Comment:
Could config be moved to the `Socialite::Server` class somehow? That'd be very useful 😬